controllers/cluster_controller.go

```diff
@@ -54,12 +55,18 @@ type AWSIamConfigReconciler interface {
 	ReconcileDelete(ctx context.Context, logger logr.Logger, cluster *anywherev1.Cluster) error
 }
 
+// ClusterValidator runs cluster level preflight validations before it goes to provider reconciler.
+type ClusterValidator interface {
+	ValidateManagementClusterName(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error
+}
+
```

Comment:
This may clash with the ClusterValidator located in the e2e tests. It may be a good idea to merge them at some point

Reply:
I think it should be fine for now because they are independent enough, don't expect the interface to clash